jscomp/ext/ext_str.mli

```diff
@@ -0,0 +1,203 @@
+(***********************************************************************)
+(*                                                                     *)
+(*                           Objective Caml                            *)
+(*                                                                     *)
+(*            Xavier Leroy, projet Cristal, INRIA Rocquencourt         *)
+(*                                                                     *)
+(*  Copyright 1996 Institut National de Recherche en Informatique et   *)
+(*  en Automatique.  All rights reserved.  This file is distributed    *)
+(*  under the terms of the GNU Library General Public License, with    *)
+(*  linking exception.                                                 *)
+(*                                                                     *)
+(***********************************************************************)
+
+(* $Id: re_str.mli,v 1.1 2002/01/16 14:16:04 vouillon Exp $ *)
+
+(** Module [Str]: regular expressions and high-level string processing *)
+
+(** {2 Regular expressions} *)
+
+type regexp
+(** The type of compiled regular expressions. *)
+
+val regexp: string -> regexp
+(** Compile a regular expression. The syntax for regular expressions
+           is the same as in Gnu Emacs. The special characters are
+           [$^.*+?[]]. The following constructs are recognized:
+    -          [.     ] matches any character except newline
```

Comment:
`str` comes with  ocaml core distribution, so you don't need copy here. But I would suggest not using it, since you only need one function which can be replaced with a string function easily